source/server/options_impl.cc

```diff
@@ -282,6 +291,33 @@ OptionsImpl::OptionsImpl(std::vector<std::string> args,
   if (!disable_extensions.getValue().empty()) {
     disabled_extensions_ = absl::StrSplit(disable_extensions.getValue(), ',');
   }
+
+  if (!stats_tag.getValue().empty()) {
+    for (const auto& cli_tag_pair : stats_tag.getValue()) {
+
+      absl::string_view::size_type colon = cli_tag_pair.find(':');
```

Comment:
I think this would have fewer subtle corner cases if you used `splitToken()` from `source/common/common/utility.h`.

Reply:
Heh -- that was more or less what this looked like in an earlier version of this PR.

I felt like that added corner cases (value contains `":"`). What corner cases does it remove?

Reply:
Oh, that's a good point; is it valid to have `:` in the value?

Reply:
> is it valid to have: in the value?

According to the implementation in this PR it is. We also have a test for it in `ValidStatsTagsCharacters`

Reply:
I think there's nothing stopping that happening now. Part of why I wanted you in particular in this was to see what sink-related limitations there are in the wild.

In particular, if we have an ipv6 address or a "host:port" construct as a tag value it will have `:` in it.

Reply:
See https://github.com/envoyproxy/envoy/blob/1c123129f91dd974e1f908748eb6128603f66590/test/server/options_impl_test.cc#L421

Reply:
To answer your question, I hate having to think about whether the offset should be +1 or not for the substr calls.

This could be `std::vector<std::string> v = absl::StrSplit(str, MaxSplits(':', 1));` I think (or very similar invocation) to avoid that. Then just check that the vector contains two elements.

Reply:
I'm OK with this -- @davinci26 had that before and I urged him off it :). But I'm OK going back to it.

Reply:
I don't mind either implementation. Let's go back to splitting then.